migrate/20250130_create_host_provider_table.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    run <<-SQL
+      CREATE TABLE host_provider (LIKE hetzner_host INCLUDING ALL);
+    SQL
+
+    alter_table(:host_provider) do
+      add_column :provider_name, :text
+    end
```

Comment:
Since it only has 2 columns, I prefer the explicit option. In addition, It should be unique by both provider and identifier, not just by the identifier like `hetzner_host`.

```suggestion
    alter_table(:host_provider) do
      foreign_key :id, :vm_host, primary_key: true, type: :uuid
      column :provider_name, :text, null: false
      column :server_identifier, :text, null: false
      index [:provider_name, :server_identifier], unique: true
    end
```